app/controllers/spree/users_controller.rb

```diff
@@ -26,16 +27,33 @@ def show
 
     # Endpoint for queries to check if a user is already registered
     def registered_email
-      user = Spree::User.find_by email: params[:email]
-      render json: { registered: user.present? }
+      registered = Spree::User.find_by(email: params[:email]).present?
+
+      if registered
+        render status: :ok, operations: cable_car.
+          inner_html(
+            "#login-feedback",
+            partial("layouts/alert", locals: { type: "alert", message: t('devise.failure.already_registered') })
+          ).
+          dispatch_event(name: "login:modal:open")
```

Comment:
I should probably do a write-up or a guide at some point for these things. Hopefully it's relatively clear and readable anyway.

The response here is using two Cable Ready Operations that are chained together: [inner_html](https://cableready.stimulusreflex.com/v/v5/reference/operations/dom-mutations#inner_html) and [dispatch_event](https://cableready.stimulusreflex.com/v/v5/reference/operations/event-dispatch#dispatch_event). You can chain as many as you like, and there are over 30 operations for doing just about any client-side DOM or browser manipulations you can imagine.

The first one (`inner_html`) is rendering a tiny partial that shows an alert message, and telling the client to overwrite the content of the `#login-feedback` element on the page with that little bit of HTML. It's a surgical micro-update where we're rendering on the server side and live-patching a tiny bit of the page's content.

The second one (`dispatch_event`) is a bit more exotic; it's telling the client to emit a custom DOM Event (`"login:modal:open"`), which the login modal listens for. This event is already used elsewhere in a couple of different places as a generic way to "ask" the login modal to open itself.